### DISCUSSION TITLE
Kafka Data Bridge configuration

### DISCUSSION THREAD

**Discussion by tommyJimmy87** [emqx/emqx-operator#665] [General]
Hi, I'm trying to setup a Kafka Bridge with a rule from EMQX to Kafka. If I set up everything from the dashboard this is working, is there any documentation about how to set it up with the operator? Is there any example/documentation? Thanks!

**Comment by Rory-Z** [comment]
Hi @tommyJimmy87 Sorry we didn't have example/documentation for now.
But I have a hack idea. After set up Kafka Bridge in dashboard, you can got all of EMQX configure in `data/cluster-overwirte.conf`, and you can copy the content you care about to the `.spec.bootstrapConfig` of EMQX CR.
Please be careful, this is not a best practice, we will improve our documentation as soon as possible.

**Comment by tommyJimmy87** [comment]
Hi @Rory-Z , the problem with the bootstrap config I think is that if I restart / re-deploy the cluster these will not being overwritten if you have a persistent volume, I, therefore, used environmental variables to create the data bridge + rules + auto subscriber. This worked out, maybe this could be helpful for others as well!

**Reply by Rory-Z** [reply]
The environmental variables can be overwrite persistent volume configure ? please tell me more details, maybe this is a bug for EMQX, because in EMQX 5.0 design, the persistent volume configure have the highest priority.

**Reply by tommyJimmy87** [reply]
That I don't know, as far as I read the cluster overwrite should have the highest, but I need to be flexible and that's why I used the variables. Every time there's a new deployment the variables will overwrite the previous value and therefore I have the new configurations in place.

**Reply by Rory-Z** [reply]
@zhongwencool Could you please take a look ？

**Reply by zhongwencool** [reply]
What is your version exactly?
Currently the configuration priority  should be `data/configs/cluster_override.conf > ENV > bootstrapconfig`
If you want the environment variables to take effect after a reboot, You must delete data/configs/*conf after stop emqx,
otherwise it's a bug.
